lib/puppet/indirector/node/write_only_yaml.rb

```diff
@@ -11,7 +11,12 @@
 # @api private
 #
 class Puppet::Node::WriteOnlyYaml < Puppet::Indirector::Yaml
-  desc "Store node information as flat files, serialized using YAML,
+  def initialize
+    Puppet.deprecation_warning('Puppet::Node::WriteOnlyYaml is deprecated and will be removed in a future release of Puppet.')
+    super
+  end
```

Comment:
Sure! I figured since we're only loading this once per indirection/indirected class we wouldn't be too spammy, but agreed - best to ensure we only do it once.